tests/core/lcp/test_credential.py

```diff
@@ -63,18 +77,23 @@ def test_getter(self, _, credential_type, method_name, expected_result):
         ) as persistent_token_create_mock:
             persistent_token_create_mock.return_value = (credential, True)
 
-            method = getattr(self._factory, method_name)
+            method = getattr(data.factory, method_name)
 
             # Act
-            result = method(self._db, self._patron)
+            result = method(session, data.patron)
 
             # Assert
             assert result == expected_result
             persistent_token_create_mock.assert_called_once_with(
-                self._db, self._data_source, credential_type, self._patron, None
+                session, data.data_source, credential_type, data.patron, None
             )
 
-    def test_get_patron_passphrase(self):
+    def test_get_patron_passphrase(
+        self, example_credential_fixture: ExampleCredentialFixture
+    ):
+        data = example_credential_fixture
+        session = data.transaction.session()
```

Comment:
This is a little different than the previous method, can we add a little inline documentation in the Database[Transaction]Fixture about having to do `.session()` in the tests?

Reply:
Yep, can do. The previous code worked in terms of `DatabaseTest`, which _kind of_ wrapped a database transaction, although did a lot of other things too. Some tests reached into the transaction and did things with the raw ORM session, and some did things with the transaction itself (although rarely).

Reply:
Is there a case where a test using the DatabaseFixture will NOT use a session?

Reply:
There are lots of cases where tests won't call `session()` directly, but they all use the session indirectly via the various methods on the fixture (like `lane()`, `work()`, etc).